cluster-app-values-schema/README.md

```diff
@@ -0,0 +1,283 @@
+        "description": "Identifies this cluster uniquely within the installation.",
+        ...
+    }
+  }
+}
+```
+
+If a description is given, additional requirements apply to the value:
+
+- Description content MUST NOT include any line breaks.
+- Descriptions MUST NOT contain formatting code like e.g. Markdown or HTML, leading or trailing white space, control characters, tabs, nor multiple whitespaces in a row.
+- Descriptions MUST use sentence case capitalization and punctuation.
+- The description SHOULD NOT repeat the property name or title. Correct: "Identifies this cluster uniquely within the installation.", incorrect: "Cluster name identifies this cluster uniquely within the installation.".
+- Descriptions SHOULD be between 50 and 200 characters long.
+- Descriptions SHOULD be written in simple language.
+
+### R6: Properties should provide examples {#r6}
+
+Each property SHOULD provide one or more examples using the `examples` keyword.
+
+Example:
+
+```json
+{
+  ...
+  "properties": {
+    "name": {
+        "type": "string",
+        "examples": ["devel001"],
+        ...
+    }
+  }
+}
+```
+
+Best practices:
+
+- Ideally, provide one example.
+- Provide more than one example if you want to highlight that very different values are accepted.
+- Provide no more than five examples
+
+TODO
+
+- We could decide to use the examples for testing purposes, replacing ci-values.yaml and the likes. In that case, we should make it a MUST requirement.
+
+### R7: Constrain values as much as possible {#r7}
+
+Property schema SHOULD explicitly restrict values as much as possible.
+
+There are several ways this requirement can/has to be fulfilled, depending on the property type and the use case.
+
+Properties of type `string`:
+
+- Use of the keywords `minLength` and `maxLength` to specify the valid length.
+- If applicable, use the `pattern` keyword to specify a regular expression for validation. This is useful for example to restrict the value to ASCII characters, or to lowercase etc.
+- Use the `format` keyword to restrict the property to a common string format, like a URL or an email address. Watch out to use only formats which are available in the JSON schema dialect used (see also R1).
+- Use `enum` if only certain known values can be valid.
+
+Numeric properties (type `number`, `integer`):
+
+- Restrict the values range using `minimum` or `exclusiveMinimum` and `maximum` or `exclusiveMaximum`.
+- Use `multipleOf` if the value has to be a multiple of a certain number.
+
+Example:
+
+```json
+"diskSizeInGb": {
+  "type": "integer",
+  "minimum": 10,
+  "maximum": 200,
+  "multipleOf": 10
+}
+```
+
+### R8: Required properties {#r8}
+
+If a property is required to be set by the creator of a cluster, it MUST be included in the `required` keyword.
+
+Properties that have default values defined in the app's `values.yaml` file MUST NOT be set as required, as here the default values are applied in case no user value is given.
```

Comment:
Regarding the discussion after the JSON schema intro lunch talk:
It seems to me that people are bothered by the fact that they have to maintain `values.yaml` and `values.schema.json`. We could work towards a solution that generates `values.yaml` from `values.schema.json`.
In that case we should mention that we use `default` for that purpose.